mlflow/cli.py

```diff
@@ -519,17 +520,32 @@ def server(
     " are not specified, data is removed for all runs in the `deleted`"
     " lifecycle stage.",
 )
-def gc(older_than, backend_store_uri, run_ids):
+@click.option(
+    "--experiment-ids",
+    default=None,
+    help="Optional comma separated list of experiments to be permanently deleted including "
+    "all of its runs. If experiment ids are not specified, data is removed for all "
+    "experiments in the `deleted` lifecycle stage.",
+)
+def gc(older_than, backend_store_uri, run_ids, experiment_ids):
     """
     Permanently delete runs in the `deleted` lifecycle stage from the specified backend store.
     This command deletes all artifacts and metadata associated with the specified runs.
     """
+    import warnings
+    from mlflow.utils.time_utils import get_current_time_millis
+
     backend_store = _get_store(backend_store_uri, None)
     if not hasattr(backend_store, "_hard_delete_run"):
         raise MlflowException(
             "This cli can only be used with a backend that allows hard-deleting runs"
         )
 
+    if not hasattr(backend_store, "_hard_delete_experiment"):
+        raise MlflowException(
+            "This cli can only be used with a backend that allows hard-deleting experiments"
+        )
```

Comment:
For backwards compatibility, can we instead skip experiment deletion if the backend doesn't define `_hard_delete_experiment` and print a warning?

Reply:
@dbczumar @harupy What should be the warning message? Do I need to say like "This cli doesn't have the backend that allows hard-deleting experiments. Update mlflow version >= {current_version}"?